tests/test_handshake.py

```diff
@@ -107,5 +108,50 @@ def test_client_handshake_data(self):
                 assert future()
                 return
 
+    @unittest.skipUnless(version_tuple >= (3, 11, -1), "requires PyMongo 3.11")
+    def test_client_handshake_saslSupportedMechs(self):
+        server = MockupDB()
+        server.run()
+        self.addCleanup(server.stop)
+
+        primary_response = OpReply('ismaster', True,
+                                   minWireVersion=2, maxWireVersion=6)
+        client = MongoClient(server.uri,
+                             username='username',
+                             password='password')
+
+        self.addCleanup(client.close)
+
+        # New monitoring sockets send data during handshake.
+        heartbeat = server.receives('ismaster')
+        heartbeat.ok(primary_response)
+
+        future = go(client.db.command, 'whatever')
```

Comment:
Should we use a command that actually exists to avoid giving the impression that using an invalid command is actually why OperationFailure is raised?

Reply:
This command never even runs so I think it's fine. "whatever" is also the same command we use in the other test in this file.